bindings/rust/s2n-tls/src/config.rs

```diff
@@ -15,7 +15,7 @@ use std::{
     sync::atomic::{AtomicUsize, Ordering},
 };
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, Eq, PartialEq)]
 pub struct Config(NonNull<s2n_config>);
```

Comment:
Why is Eq needed now?

Reply:
This is error I got without it:

```
> cargo clippy --manifest-path bindings/rust/Cargo.toml --all-targets -- -D warnings                      

Checking s2n-tls v0.0.11 (/Volumes/workplace/s2n-tls/bindings/rust/s2n-tls)
error: you are deriving `PartialEq` and can implement `Eq`
  --> s2n-tls/src/config.rs:18:17
   |
18 | #[derive(Debug, PartialEq)]
   |                 ^^^^^^^^^ help: consider deriving `Eq` as well: `PartialEq, Eq`
   |
   = note: `-D clippy::derive-partial-eq-without-eq` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq

error: could not compile `s2n-tls` due to previous error
```

Reply:
This lint is super annoying and we should really ignore it for public types. Adding `Eq` will potentially make it more difficult to add fields or variants in the future that don't implement `Eq`.

Reply:
Gotcha, will remove

Reply:
you can see this for ignoring clippy warnings: https://github.com/rust-lang/rust-clippy#configuration

Reply:
I am not sure Eq gives us much and we can always add it. (2 way door)

However if we add the Eq accidentally then we might need to remove it and cant because of backwards compatibility (1 way door)

I think we should add the exception lint to clippy.toml and opt out of this lint entirely. That way clippy is not telling us to do the potentially more dangerous thing.